### PR TITLE
HMAN-203 changed GET /hearingActuals/ the individualDetails to a single object

### PR DESCRIPTION
**Pull request by padmapriyanalam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-203

### Change description ###

GET /hearingActuals/ the individualDetails is an ARRAY not a single obj

Changed Individual details mapping to 1-1 from Hearing Party instead of 1-many

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java [hunk 1]
```diff
@@ -127,25 +129,19 @@ private void setActualDayParties(ActualHearingDayEntity actualHearingDayEntity,
...
-                    individualDetails.setLastName(individualDetailEntity.getLastName());
-                    individualDetailsList.add(individualDetails);
-
-                } else {
-                    ActualOrganisationDetails organisationDetails = new ActualOrganisationDetails();
-                    organisationDetails.setName(individualDetailEntity.getPartyOrganisationName());
-                    organisationDetailsList.add(organisationDetails);
-                }
+            ActualIndividualDetails individualDetails = new ActualIndividualDetails();
+            ActualOrganisationDetails organisationDetails = new ActualOrganisationDetails();
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java:133]: I don't think we should be creating these objects unless we actually need them. We can just create them in the `if`/`else` statement, only when required?

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java [hunk 2]
```diff
@@ -127,25 +129,19 @@ private void setActualDayParties(ActualHearingDayEntity actualHearingDayEntity,
...
+            if (individualDetailEntity.getPartyOrganisationName() == null) {
+                individualDetails.setFirstName(individualDetailEntity.getFirstName());
+                individualDetails.setLastName(individualDetailEntity.getLastName());
+            } else {
+                organisationDetails.setName(individualDetailEntity.getPartyOrganisationName());
             }
-            actualDayParty.setActualIndividualDetails(individualDetailsList);
-            actualDayParty.setActualOrganisationDetails(organisationDetailsList);
+            actualDayParty.setActualIndividualDetails(individualDetails);
+            actualDayParty.setActualOrganisationDetails(organisationDetails);
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java:144]: Similar to previous comment, we can set this in the appropriate `if`/`else`.

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java [hunk 3]
```diff
@@ -237,17 +233,14 @@ private OrganisationDetails setOrganisationDetails(HearingPartyEntity hearingPar
         return organisationDetails;
     }
 
-    private ArrayList<IndividualDetails> setIndividualDetails(HearingPartyEntity hearingPartyEntity) {
-        ArrayList<IndividualDetails> individualDetailsArrayList = new ArrayList<>();
+    private IndividualDetails setIndividualDetails(HearingPartyEntity hearingPartyEntity) {
+        IndividualDetails individualDetails = new IndividualDetails();
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java:237]:
```suggestion
        IndividualDetails individualDetails = null;
```
Should probably instantiate inside the `if`, so it's only when required?

**Review by markdathornehmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java (hunk 1 above)
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java]: changed

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java (hunk 2 above)
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java:146]: changed

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java (hunk 3 above)
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java]: changed

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapper.java [hunk 4]
```diff
@@ -156,22 +156,18 @@ private OrganisationDetails setOrganisationDetails(HearingPartyEntity hearingPar
         return organisationDetails;
     }
 
-    private ArrayList<IndividualDetails> setIndividualDetails(HearingPartyEntity hearingPartyEntity) {
-        ArrayList<IndividualDetails> individualDetailsArrayList = new ArrayList<>();
+    private IndividualDetails setIndividualDetails(HearingPartyEntity hearingPartyEntity) {
+        IndividualDetails individualDetails = new IndividualDetails();
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapper.java]: changed

**Review by danlysiak — APPROVED** [review]

**Pull request merged**